community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogNeoCommandReaderV2_1.java

```diff
@@ -309,7 +310,7 @@ public boolean visitRelationshipGroupCommand( Command.RelationshipGroupCommand c
             {
                 throw new IOException( "Illegal in use flag: " + inUseByte );
             }
-            int type = channel.getShort();
+            int type = shortToUnsignedInt( channel.getShort() );
```

Comment:
It is a software upgrade but not the store format upgrade :)
When 2.2.10 database starts on 2.2.4 non-clean store it uses `V2_2_4` reader to real 2.2.4 log during recovery. Later it uses it's own `V2_2_10` log reader. Here there is no store upgrade involved.